File: backend/app/schemas/trip.py
from datetime import date

from pydantic import BaseModel


class ExtractRequest(BaseModel):
    input: str


class ExtractedTrip(BaseModel):
    destination: str
    destination_country: str
    start_date: str
    end_date: str
    duration_days: int
    purpose: str
    special_scenarios: list[str] = []
    group_size: int = 1
    group_description: str = "1人"


class CreateTripRequest(BaseModel):
    user_input: str
    destination: str
    country: str | None = None
    start_date: str
    end_date: str
    duration_days: int
    purpose: str
    special_scenarios: list[str] = []
    group_size: int = 1
    group_description: str = "1人"


class TripResponse(BaseModel):
    id: int
    user_input: str
    destination: str
    country: str | None
    start_date: str
    end_date: str
    duration_days: int
    purpose: str
    special_scenarios: list[str]
    group_size: int
    group_description: str | None
    status: str

    model_config = {"from_attributes": True}

File: frontend/src/api/trip.ts
import client from "./client";
import type {
  ExtractedTrip,
  CreateTripResponse,
  TripResponse,
  ChecklistItem,
  ProgressData,
} from "../types";

export async function extractTripInfo(input: string): Promise<ExtractedTrip> {
  const res = await client.post("/api/trips/extract", { input });
  return res.data;
}

export async function createTrip(trip: {
  user_input: string;
  destination: string;
  country?: string;
  start_date: string;
  end_date: string;
  duration_days: number;
  purpose: string;
  special_scenarios: string[];
  group_size: number;
  group_description: string;
}): Promise<CreateTripResponse> {
  const res = await client.post("/api/trips", trip);
  return res.data;
}

export async function getTrip(tripId: number): Promise<{
  trip: TripResponse;
  checklist: ChecklistItem[];
  progress: ProgressData;
}> {
  const res = await client.get(`/api/trips/${tripId}`);
  return res.data;
}

export async function listTrips(): Promise<
  { id: number; destination: string; start_date: string; end_date: string }[]
> {
  const res = await client.get("/api/trips");
  return res.data;
}
